vortex-dtype/src/dtype.rs

```diff
@@ -138,6 +138,14 @@ impl DType {
             _ => None,
         }
     }
+
+    /// Get the inner dtype if `self` is a `ListDType`, otherwise `None`
+    pub fn as_list(&self) -> Option<(&DType, &Nullability)> {
```

Comment:
do we need to return outer nullability here? e.g. as_struct doesn't do this

Reply:
I have removed it, but it does seem strange that as_struct, doesn't I think it is that vortex top level compute functions apply nullability.